source/4.0/learn/introduction/index.html.md

```diff
@@ -1,24 +1,148 @@
+is this idea that it's easy to:
+
+1. Map objects to database tables one-to-one (the
+[ActiveRecord](https://en.wikipedia.org/wiki/Active_record_pattern) design
+pattern); or
+
+2. Introduce machinery to translate between objects and persistence structures
+(the [DataMapper](https://en.wikipedia.org/wiki/Data_mapper_pattern) pattern)
+
+Both strategies tend to start out fine, but can quickly become cumbersome as an
+application transitions to a medium-to-large application.
+
+ActiveRecord style mapping has the benefit of aiding in rapid prototyping works
+great in basic CRUD scenarios but it limits application modeling to what's
+convenient for the database. Entities will tend to map one-to-one with tables
+which leads to knowledge of the persistence structure infecting the application
+domain leading to change resistant software.
+
+The DataMapper pattern, while one step better than ActiveRecord style mapping,
+still has its focus trained onto creating and managing objects. The pattern
+solves the one-to-one mapping issue but in doing so creates a host of others.
+The whole apparatus needed to manage all of those objects inevitably leads to
+performance issues and global state tracking which results in developers
+bypassing the ORM entirely.
+
+The problems with ORMs are numerous and the above issues only begin to scratch
+the surface. If you're interested in further reading on the subject, we suggest:
+
+* [ORM Hate by Martin Fowler](https://martinfowler.com/bliki/OrmHate.html)
+* [The Vietnam of Computer Science by Ted Neward](http://blogs.tedneward.com/post/the-vietnam-of-computer-science/)
+
+## Why use ROM
+
+ROM provides an alternative way of handling persistence and related concerns.
+It focuses on *simplicity* by providing enough *abstractions* to help you
+efficiently turn your raw data into meaningful information. 
+
+While many ORMs focus on objects and state tracking, ROM focuses on data and
+transformations. Users of ROM implement `Relations` which map one-to-one with
+datasets (eg: tables). Then using the relations you can associate them with
+other relations and query the dataset using features offered by the
+datastore. Once raw data has been loaded it gets coerced into configured
+datatypes and from there can mapped into whatever format is needed by the
+application domain including custom type objects, helpful ROM Structs or plain
+old ruby hashes.
+
+The important concept above is during the entire process there is no dirty
+tracking, no identity management, no mutable state or anything else. Just *pure*
+data being loaded and mapped as result of a **direct** request made from the
+application domain. Data can be persisted in ways that take advantage of the
+features provided by the datastore and the application domain can receive that
+data in any form it needs. Furthermore you get the added benefits of:
+
+  * decoupling the application from the persistence layer without sacrificing
+  * flexibility, and
```

Comment:
looks like this should be a single point instead of two

Reply:
Whoops it was supposed to be a single point, my text-reflow in Atom screwed up.